src/contract/contract_function_parameters.rs

```diff
@@ -1013,6 +1035,17 @@ where
     out_bytes
 }
 
+fn encode_array_of_32_byte<T: IntoBytes32>(elements: T) -> Vec<u8> {
+    let slice = elements.get_bytes_32();
+    if slice.len() > 32 {
+        panic!("32 bytes exceeded in contract function call")
+    }
+
+    let mut new_bytes = slice;
+    right_pad_32_bytes(&mut new_bytes);
```

Comment:
this function doesn't take `&mut`, it takes an `&[u8]` and returns a `[u8; 32]`,

Reply:
You might be looking at `left_pad_32_bytes()`. `right_pad` Takes `&mut Vec<8>` and returns nothing.

Reply:
```
fn right_pad_32_bytes(bytes: &[u8]) -> [u8; 32] {
    let mut result = [0_u8; 32];
    result[..bytes.len()].copy_from_slice(bytes);
    result
}
```

Reply:
Are we looking at the same one? https://github.com/hashgraph/hedera-sdk-rust/blob/0ca7fc8e05264107ff69c7521e63eec546c0761c/src/contract/contract_function_parameters.rs#L984-L991

Reply:
uh, no we were not.